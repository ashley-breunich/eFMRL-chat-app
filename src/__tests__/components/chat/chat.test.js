import React from 'react';
import renderer from 'react-test-renderer';
import Chat from 'components/chat/chat.js';

describe('<Chat />', () => {
  it('Chat is alive at application start - form', () => {
    let app = shallow(<Chat />);
    expect(app.find('div').exists()).toBeTruthy();
  });

  it('Chat is alive at application start - button', () => {
    let app = shallow(<Chat />);
    expect(app.find('button').exists()).toBeTruthy();
  });

  it('Chat is alive at application start - h2', () => {
    let app = shallow(<Chat />);
    expect(app.find('h2').exists()).toBeTruthy();
  });

  it('Chat is alive at application start - form', () => {
    let app = shallow(<Chat />);
    expect(app.find('form').exists()).toBeTruthy();
  });

  it('Chat is alive at application start - input', () => {
    let app = shallow(<Chat />);
    expect(app.find('input').exists()).toBeTruthy();
  });

  it('Chat is alive at application start - ul', () => {
    let app = shallow(<Chat />);
    expect(app.find('ul').exists()).toBeTruthy();
  });

  it('Moniker renders correctly', () => {
    const header = renderer.create(<Chat />).toJSON();
    expect(header).toMatchSnapshot();
  });
});
