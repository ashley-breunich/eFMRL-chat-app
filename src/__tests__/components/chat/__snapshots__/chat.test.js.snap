// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Chat /> Moniker renders correctly 1`] = `
<form
  className="loginForm"
  onSubmit={[Function]}
>
  <input
    className="loginInput"
    name="typedInput"
    onChange={[Function]}
    placeholder="What is your name?"
  />
  <p
    className="loginError"
  >
    
  </p>
</form>
`;
